source/manual/move-apps-between-servers.html.md

```diff
@@ -0,0 +1,103 @@
+---
+owner_slack: "#2ndline"
+title: Move apps between servers
+section: Environments
+layout: manual_layout
+parent: "/manual.html"
+last_reviewed_on: 2018-03-02
+review_in: 6 months
+---
+
+Most frontend and backend apps on GOV.UK share a small number of servers. In some circumstances, apps may use more than their share of resources and may affect other apps on the same server. In these cases, apps can be moved to their own servers using the appropriate steps for either Carrenza or AWS.
+
+## Carrenza
+
+### Create the new servers (if required)
+
+If you're moving an app to new servers, start by creating those server.
+
+1. Add [vCloud configuration][govuk-provisioning] for production and staging (you can choose any IP addresses that are not currently used as long as they use the same prefix as others in the same file).
+1. Add the IP addresses for the new servers and a new Puppet node class in [govuk-puppet].
+1. Deploy puppet in staging and production.
+1. In staging, SSH to the puppetmaster and run a loop to sign SSL certificates for the new servers as they're created:
+    ```bash
+    $ ssh puppetmaster-1.management.staging
+    $ while true; do sudo puppet cert sign --all; sleep 10; done
+    ```
+1. Run the [Launch VMs][launch-vms-jenkins] Jenkins job, using the [Carrenza staging username and password][carrenza-credentials], to create the new servers.
+1. Once the job has completed, terminate the loop on the puppetmaster.
+1. Re-run the loop and Jenkins job in production.
+
+[govuk-provisioning]: https://github.com/alphagov/govuk-provisioning/pull/41
+[govuk-puppet]: https://github.com/alphagov/govuk-puppet/pull/7294
+[launch-vms-jenkins]: https://deploy.staging.publishing.service.gov.uk/job/Launch_VMs/
+[carrenza-credentials]: https://github.com/alphagov/govuk-secrets/blob/master/pass/2ndline/carrenza/vcloud-staging.gpg
+
+### Add the relevant app(s) to the new servers
+
+Once the servers are created, they will run puppet to apply relevant configuration, but there will be no apps.
+
+1. Change the [hieradata][] to add the relevant app(s) to the new servers.
+1. Deploy puppet in staging and production.
+1. To speed up the process, you can run puppet manually on the new servers to deploy the apps.
```

Comment:
Might be worth breaking this up into deploy on staging, then run manually, then deploy prod. As it sounds a bit like do both deploys rather than more cautious staging, then check works, then to prod.

Reply:
Yup, agree, I've done this for all the stages.